e2e/vue3-vite-e2e/project.json

```diff
@@ -7,7 +7,8 @@
       "executor": "@nrwl/nx-plugin:e2e",
       "options": {
         "target": "vue3-vite:build",
-        "jestConfig": "e2e/vue3-vite-e2e/jest.config.ts"
+        "jestConfig": "e2e/vue3-vite-e2e/jest.config.ts",
+        "runInBand": true
```

Comment:
From what I observed, Jest runs tests in parallel by default, so the `ensureNxProject` in the `beforeEach` of each `.spec.ts` is run one after another, causing the tests to fail in a strange way.
In the end, it appears that using the [`runInBand`](https://nx.dev/packages/nx-plugin/executors/e2e#runinband) option is the simplest solution, however, **it greatly reduces the performance of e2e tests**! :confused:

I've tried using `setupFiles` in `jest.config.ts` and call `beforeAll` in it, but it didn't work as expected; I didn't manage to make it run only once.

Reply:
It would be great if we can find a way to run `ensureNxProject` only once before any test (`.spec.ts`) is run.
But at the same time we also want it to work with the `--test-file=` option during development.

Reply:
Here's the code I use in `jest.setup.ts`:

```ts
let [done, processing] = [false, false]

beforeAll(async () => {
  if (processing) {
    await waitForSetup()
  } else if (!done) {
    // I'm not sure if the following assumption is correct:
    // 
    // Even if tests are run in parallel by Jest, those run are still started
    // sequentially, it's just that multiple runs can happen at the same time.
    // 
    // So I expect the code in this block will only be run in the very first run.
    processing = true
    ensureNxProject(...)
    done = true
    processing = false
  }
})

const waitForSetup = (): Promise<void> => {
  return new Promise((resolve) => {
    // Use `setInterval` to periodically check if `done` has been
    // changed to `true`; if yes, then call `resolve` and `clearInterval`.
  })
}
```

It probably makes no sense to you at all; I expect this to work, but it didn't. 😅

Reply:
I will see what I can do!

Jest does have a feature for [global setup/teardown](https://jestjs.io/docs/configuration#globalsetup-string), but I'm not sure it will solve the problem. Since `ensureNxProject` always create a tmp E2E project folder with the same path, the tests will interfere with each other.

Looking at the code for `ensureNxProject`, I think it would be straightforward to modify it to pass a custom test path, but they don't make it easy to adjust without sending a PR. I will see if jest's mocking capabilities can handle it, otherwise we can copy/paste the Nx code and make the changes ourselves. If Nx fixes [this issue](https://github.com/nrwl/nx/issues/4601) it would resolve our problem, but it seems like there wasn't much interest.